client/src/components/sources/sourcesToolbar.js

```diff
@@ -0,0 +1,282 @@
+
+class SourcesToolbar extends React.Component {
+  constructor() {
+    super();
+
+    bindMethods(this, [
+      'updateCurrentValue',
+      'onValueKeyPress',
+      'selectFilterType',
+      'filterValueSelected',
+      'filterAdded',
+      'removeFilter',
+      'clearFilters',
+      'updateCurrentSortType',
+      'toggleCurrentSortDirection'
+    ]);
+  }
+
+  componentDidMount() {
+    const { filterType, sortType } = this.props;
+
+    if (!filterType) {
+      this.selectFilterType(SourceFilterFields[0]);
+    }
+
+    if (!sortType) {
+      this.updateCurrentSortType(SourceSortFields[0]);
+    }
+  }
+
+  filterAdded(field, value) {
+    let filterText = '';
+    if (field.title) {
+      filterText = field.title;
+    } else {
+      filterText = field;
+    }
+    filterText += ': ';
+
+    if (value.title) {
+      filterText += value.title;
+    } else {
+      filterText += value;
+    }
+
+    let filter = { field: field, value: value, label: filterText };
+    Store.dispatch({
+      type: dispatchTypes.ADD_FILTER,
+      filter
+    });
+  }
+
+  selectFilterType(filterType) {
+    Store.dispatch({
+      type: dispatchTypes.SET_FILTER_TYPE,
+      filterType
+    });
+  }
+
+  filterValueSelected(newFilterValue) {
+    const { filterType } = this.props;
+
+    let filterValue = newFilterValue;
+    Store.dispatch({
+      type: dispatchTypes.SET_FILTER_VALUE,
+      filterValue
+    });
+    if (newFilterValue) {
+      this.filterAdded(filterType, newFilterValue);
+    }
+  }
+
+  updateCurrentValue(event) {
+    let filterValue = event.target.value;
+    Store.dispatch({
+      type: dispatchTypes.SET_FILTER_VALUE,
+      filterValue
+    });
+  }
+
+  onValueKeyPress(keyEvent) {
+    const { filterValue, filterType } = this.props;
+
+    if (keyEvent.key === 'Enter' && filterValue && filterValue.length > 0) {
+      this.filterAdded(filterType, filterValue);
+      keyEvent.stopPropagation();
+      keyEvent.preventDefault();
+    }
+  }
+
+  removeFilter(filter) {
+    Store.dispatch({
+      type: dispatchTypes.REMOVE_FILTER,
+      filter
+    });
+  }
+
+  clearFilters() {
+    Store.dispatch({
+      type: dispatchTypes.CLEAR_FILTERS
+    });
+  }
+
+  updateCurrentSortType(sortType) {
+    Store.dispatch({
+      type: dispatchTypes.SET_SORT_TYPE,
+      sortType
+    });
+  }
+
+  toggleCurrentSortDirection() {
+    Store.dispatch({
+      type: dispatchTypes.TOGGLE_SORT_ASCENDING
+    });
+  }
+
+  renderFilterInput() {
+    const { filterType, filterValue } = this.props;
+    if (!filterType) {
+      return null;
+    }
+
+    if (filterType.filterType === 'select') {
+      return (
+        <Filter.ValueSelector
+          filterValues={filterType.filterValues}
+          currentValue={filterValue}
+          placeholder={filterType.placeholder}
+          onFilterValueSelected={this.filterValueSelected}
+        />
+      );
+    } else {
+      return (
```

Comment:
Optional: Same comment about the ```else```s with ```return```s, may be able to get away with out them and the encompassing curly braces